hazelcast/src/main/java/com/hazelcast/replicatedmap/impl/record/KeySetIteratorFactory.java

```diff
@@ -48,6 +49,18 @@ public boolean hasNext() {
 
             @Override
             public K next() {
+                if (entry == null) {
+                    while (iterator.hasNext()) {
```

Comment:
I couldn't understand how this while loop works. Shouldn't it stop when a valid entry is found?